contracts/discovery/IGNS.sol

```diff
@@ -6,20 +6,26 @@ interface IGNS {
     // -- Pool --
 
     struct SubgraphData {
-        uint256 vSignal; // The token of the subgraph-deployment bonding curve
+        uint256 vSignal; // The sum of all deployment vSignal
         uint256 nSignal; // The token of the subgraph bonding curve
         mapping(address => uint256) curatorNSignal;
         bytes32 subgraphDeploymentID;
         uint32 reserveRatio;
         bool disabled;
         uint256 withdrawableGRT;
+        mapping(uint256 => Version) versions;
```

Comment:
The implementation using the top-level mapping required more storage assignments and the code was harder to reason about. Having the mapping available in the struct allowed us use the `SubgraphData` as the single entry point for versions.

You can view what it would look like in the initial commit: https://github.com/graphprotocol/contracts/pull/526/commits/d03b31b5c768f5ce4cecafddbce4fe7f54c3fc67

Since we are already storing a mapping in this struct, we might as well keep with the pattern.